File: core/src/androidTest/java/com/emarsys/core/provider/timestamp/TimestampProviderTest.java
package com.emarsys.core.provider.timestamp;

import com.emarsys.testUtil.TimeoutUtils;

import org.junit.Assert;
import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.TestRule;

public class TimestampProviderTest {

    @Rule
    public TestRule timeout = TimeoutUtils.getTimeoutRule();

    @Test
    public void testProvideTimestamp_returnsTheCurrentTimestamp() {
        long before = System.currentTimeMillis();
        long actual = new TimestampProvider().provideTimestamp();
        long after = System.currentTimeMillis();

        Assert.assertTrue(before <= actual);
        Assert.assertTrue(actual <= after);
    }

}

File: core/src/androidTest/java/com/emarsys/core/provider/timestamp/TimestampProviderTest.kt
package com.emarsys.core.provider.timestamp

import io.kotest.matchers.shouldBe
import org.junit.Test

class TimestampProviderTest  {
    @Test
    fun testProvideTimestamp_returnsTheCurrentTimestamp() {
        val before = System.currentTimeMillis()
        val actual = TimestampProvider().provideTimestamp()
        val after = System.currentTimeMillis()
        (before <= actual) shouldBe true
        (actual <= after) shouldBe true
    }
}